integrations/GA/browser.js

```diff
@@ -119,4 +696,260 @@ class GA {
   }
 }
 
-export { GA };
\ No newline at end of file
+/**
+ * Map google's custom dimensions, metrics & content groupings with `obj`.
+ *
+ * Example:
+ *
+ *      metrics({ revenue: 1.9 }, { { metrics : { revenue: 'metric8' } });
+ *      // => { metric8: 1.9 }
+ *
+ *      metrics({ revenue: 1.9 }, {});
+ *      // => {}
+ */
+
+function metrics(obj, dimensions, metrics, contentGroupings) {
+  var ret = {};
+
+  each([metrics, dimensions, contentGroupings], function (group) {
+    each(group, function (prop, key) {
+      var value = obj[prop];
+      if (is.boolean(value)) value = value.toString();
+      if (value || value === 0) ret[key] = value;
+    });
+  });
+
+  return ret;
+}
+function formatValue(value) {
+  if (!value || value < 0) return 0;
+  return Math.round(value);
+}
+function setCustomDimenionsAndMetrics(props, inputs) {
+  var ret = {};
+  var dimensionsArray = {};
+  for (let val of inputs.dimensions) {
+    dimensionsArray[val.from] = val.to;
+  }
+  var metricsArray = {};
+  for (let val of inputs.metrics) {
+    metricsArray[val.from] = val.to;
+  }
+  var contentGroupingsArray = {};
+  for (let val of inputs.contentGroupings) {
+    contentGroupingsArray[val.from] = val.to;
+  }
+  var custom = metrics(
+    props,
+    dimensionsArray,
+    metricsArray,
+    contentGroupingsArray
+  );
+  if (Object.keys(custom).length) {
+    if (inputs.setAllMappedProps) {
+      ga("set", custom);
+    } else {
+      each(custom, function (key, value) {
+        ret[key] = value;
+      });
+      return ret;
+    }
+  }
+}
+
+// Return the path based on `properties` and `options`
+
+function path(properties, includeSearch) {
+  if (!properties) return;
+  var str = properties.path;
+  if (includeSearch && properties.search) str += properties.search;
+  return str;
+}
+
+//Creates a track out of product properties.
+
+function createProductTrack(rudderElement, properties) {
+  var props = properties || {};
+  props.currency =
+    properties.currency || rudderElement.message.properties.currency;
+  return { properties: props };
+}
+
+// Loads ec.js (unless already loaded)
+function loadEnhancedEcommerce(rudderElement, a) {
+  if (a === 0) {
```

Comment:
not required anymore right?

ec plugin is loaded in init now

Reply:
that was for e commerce and this enhanced. this can be set only depending on the type of event and not any input from user.

Reply:
can we load only one of ecommerce or enhanced ecommerce  at the time of init?
what will happen if we always load ecommerce related plugin and only when the UI input in enhanced we load that.